### PR TITLE
Fix for issue #329

### PR DESCRIPTION
**Pull request by s8sg**
We need to Enable field data on all the text fields for searching, in ES 5.0 Strings are mapped to text fields where the field data is false by default. 

nxparse.py: fileddata : true is being set at mapping for: whitelisted, server, uri, zone, ip

(*Note: string fields is being depricated by text and keyword at ES 5.0)

[Source](https://www.elastic.co/guide/en/elasticsearch/reference/current/fielddata.html)

### CODE REVIEW

**Comment by buixor** [comment]
Hello,

Thanks for the MR :)
Would you mind making the ES5 flag being read from the configuration file ?

Best regards,

**Comment by mguesdon** [comment]
Hi,

Thank you, it fix the problem !

About  ES5_ENABLE=True: may be a better solution would be to query elasticsearch to get the version ? 
curl -XGET 'localhost:9200'
{
  "name" : "naxsi-1",
  "cluster_name" : "naxsi",
  "cluster_uuid" : "8ER3_mYPQbun4c7L6hps9w",
  "version" : {
    "number" : "5.0.1",
    "build_hash" : "080bb47",
    "build_date" : "2016-11-11T22:08:49.812Z",
    "build_snapshot" : false,
    "lucene_version" : "6.2.1"
  },
  "tagline" : "You Know, for Search"
}

**Comment by s8sg** [comment]
@buixor I have updated the patch.  
The ES version is being read from the con, which is already set as mandatory.

**Comment by buixor** [comment]
Sounds cool :)

If the elasticsearch-py can query it, it would be awesome, can you check that ?
It would allow to merge blindly !

cheers,

**Comment by buixor** [comment]
awesome, thank you very much :)

In the meanwhile, do you want to give a try at the **new** nxtool version ? 

https://github.com/nbs-system/nxtool-ng

We are going to switch to this one soon.

File: nxapi/nxtool.py [hunk 1]
```diff
@@ -160,6 +156,14 @@ def get_filter(arg_filter):
     use_ssl = False
     
 es = elasticsearch.Elasticsearch(cfg.cfg["elastic"]["host"], use_ssl=use_ssl)
+# Get ES version from the client and avail it at cfg
+es_version =  es.info()['version'].get('number', None)
+if es_version is not None:
+    cfg.cfg["elastic"]["version"] = es_version.split(".")[0]
```
Comment (s8sg) [inline nxapi/nxtool.py]: Currently only version major is being treated. It is added in `cfg` to make it globally available. Version specific handling might be necessary for other functions.

**Comment by s8sg** [comment]
@buixor Glad to contribute. I would love to check the  new nxtool version. Thanks :)

File: nxapi/nxtool.py (hunk 1 above)
Comment (mguesdon) [inline nxapi/nxtool.py]: Great ! Thank you !

A little problem: v5 is not handled in nxtransform.py (search for "Unspecified ES version"), so a "./nxtool.py -c nxapi.json  -x --colors" fail with "Unknown / Unspecified ES version in nxapi.json : 5" :-)

Comment (s8sg) [inline nxapi/nxtool.py]: :( Version definite checking is being done to many places. Its not yet ready for ES 5. @buixor

Comment (mguesdon) [inline nxapi/nxtool.py]: OK, thx !

Comment (s8sg) [inline nxapi/nxtool.py]: @mguesdon Could you please review the patch #334

**Pull request merged**